dataset/test/dataset_test.cpp

```diff
@@ -253,30 +253,6 @@ TEST(DatasetTest, const_iterators_return_types) {
   ASSERT_TRUE((std::is_same_v<decltype(*d.end()), DataArray>));
 }
 
-TEST(DatasetTest, construct_from_view) {
-  DatasetFactory3D factory;
-  const auto dataset = factory.make();
-  const Dataset view(dataset);
-  Dataset from_view(view);
-  ASSERT_EQ(from_view, dataset);
-}
-
-TEST(DatasetTest, construct_from_slice) {
-  DatasetFactory3D factory;
-  const auto dataset = factory.make();
-  const auto slice = dataset.slice({Dim::X, 1});
-  Dataset from_slice(slice);
-  ASSERT_EQ(from_slice, dataset.slice({Dim::X, 1}));
-}
-
-TEST(DataArrayTest, construct_from_slice) {
-  DatasetFactory3D factory;
-  const auto dataset = factory.make();
-  const auto slice = dataset["data_xyz"].slice({Dim::X, 1});
-  DataArray from_slice(slice);
-  ASSERT_EQ(from_slice, dataset["data_xyz"].slice({Dim::X, 1}));
-}
```

Comment:
Do we have equivalent tests of `copy` from slices? If not, maybe preserve this (using `copy`instead of constructors)?

Reply:
Didn't find any. I added those two tests back in.

But I did not see any slicing tests for DataAray,

Reply:
That is historic, since `DataArray` used to be based on `Dataset`... we should definitely add some, so open card in the project board.